src/test.rs

```diff
@@ -343,3 +344,20 @@ fn check_test_environment() {
     #[cfg(target_os = "linux")]
     fn test_iface() {}
 }
+
+#[test]
+fn test_get_network_interfaces_ips() {
+    test_iface_ips();
```

Comment:
I think the right way to test this is dependency injection, replacing `libc::getifaddrs()` then testing that `get_network_interfaces()` returns the correct set of `NetworkInterface`s. This is probably overkill for this fix though - if you want to work on it, I'd love a proper test suite for it - if not I'll happily accept the pull sans-test case (I don't think it's a good idea to include the current test case).